core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java

```diff
@@ -93,11 +107,57 @@ public void build(ImageConfiguration imageConfig) throws Fabric8ServiceException
 
             // Create a file with generated image streams
             addImageStreamToFile(getImageStreamFile(config), imageName, client);
+        } catch (Fabric8ServiceException e) {
+            throw e;
         } catch (Exception ex) {
             throw new Fabric8ServiceException("Unable to build the image using the OpenShift build service", ex);
         }
     }
 
+    protected File createBuildArchive(ImageConfiguration imageConfig) throws Fabric8ServiceException {
+        // Adding S2I artifacts such as environment variables in S2I mode
+        ArchiverCustomizer customizer = getS2ICustomizer(imageConfig);
+
+        try {
+            // Create tar file with Docker archive
+            File dockerTar;
+            if (customizer != null) {
+                dockerTar = dockerServiceHub.getArchiveService().createDockerBuildArchive(imageConfig, config.getDockerMojoParameters(), customizer);
+            } else {
+                dockerTar = dockerServiceHub.getArchiveService().createDockerBuildArchive(imageConfig, config.getDockerMojoParameters());
+            }
+            return dockerTar;
+        } catch (MojoExecutionException e) {
+            throw new Fabric8ServiceException("Unable to create the build archive", e);
+        }
+    }
+
+    private ArchiverCustomizer getS2ICustomizer(ImageConfiguration imageConfiguration) throws Fabric8ServiceException {
+        try {
+            if (imageConfiguration.getBuildConfiguration() != null && imageConfiguration.getBuildConfiguration().getEnv() != null) {
+                final File environmentFile = new File(config.getBuildDirectory(), "s2i-env-" + imageConfiguration);
```

Comment:
Do you rely here on the toString() from the imageConfiguration (i.e. that it uses the default with a random memory address) ? I would do this is in a more controlled manner, also to avoid unexpected chars in the filename).

Reply:
Uh, thanks for spotting this :+1:

Probably a refactoring issue...